miner/worker.go

```diff
@@ -860,6 +863,9 @@ func (w *worker) commitNewWork(interrupt *int32, noempty bool, timestamp int64)
 	}
 
 	num := parent.Number()
+	if num.Uint64() <= w.checkpoint {
```

Comment:
If it's in private network without any checkpoint, the checkpoint number here is 0.
If it's the first block to mine(the parent is genesis), the mining operation will be aborted.